ntfy/backends/xmpp.py

```
@@ -0,0 +1,97 @@
+
+        self.add_event_handler("session_start", self.start)
+
+    def start(self, event):
+
+        self.send_presence()
+        self.get_roster()
+        msg_args = {'mto': self.recipient,
+                    'msubject': self.title,
+                    'mbody': self.msg}
+        if self.mtype:
+            msg_args['mtype'] = self.mtype
+
+        self.send_message(**msg_args)
+
+        self.disconnect(wait=True)
+
+def notify(title, message, config, recipient=None, mtype=None,
+           manual_server=None, ssl_version=None, path_to_certs=None,
+           **kwargs):
+    """
+    Required config keys:
+        * jid
+        * password
+        * recipient (or supply via kw)
+
+    Optional config or kw:
+        * mtype ('chat' required for Google Hangouts)
+        * manual_server
+        * ssl_version
+        * path_to_certs
+
+    To verify the SSL certificates offered by a server:
+    path_to_certs = "path/to/ca/cert"
+
+    OpenFire server may need to adjust the SSL version ie
+    ssl_version = ssl.PROTOCOL_SSLv3
+
+    If you do not have the dnspython library installed, you will need
+    to manually specify the name of the server if it does not match
+    the one in the JID. For example, to use Google Talk you would
+    need to use:
+    manual_server = ('talk.google.com', 5222)
+    """
+
+    jid = config['jid']
+    password = config['password']
+    recipient = recipient or config['recipient']
+    mtype = mtype or config.get('mtype', None)
+    xmpp_bot = NtfySendMsgBot(jid, password, recipient,
+                              title, message, mtype)
+
+    # NOTE: Below plugins weren't needed for Google Hangouts
+    # but probably useful (were in original sleekxmpp example)
+    # xmpp_bot.register_plugin('xep_0030') # Service Discovery
+    # xmpp_bot.register_plugin('xep_0199') # XMPP Ping
+
+    path_to_certs = path_to_certs or config.get('path_to_certs', None)
+    if path_to_certs and os.path.isdir(path_to_certs):
```

Should we raise an exception if it's supplied and isn't a directory?